src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -418,15 +420,34 @@ private async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
         public override async Task RunAsync()
         {
             await PreRunConditions();
+            var isVerbose = VerboseLogging.HasValue && VerboseLogging.Value;
 
-            var isCurrentProcessNet6Build = RuntimeInformation.FrameworkDescription.Contains(Net6FrameworkDescriptionPrefix);
-            if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            // If --runtime param is set, handle runtime param logic; otherwise we infer the host to launch on startup
+            if (HostRuntime is not null)
```

Comment:
All the logic inside this `if` block should be done only if the CurrentWorkerRuntime is "dotnet". Should we check that here first and move on?